portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -56,11 +56,18 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       String commentEntity = (String) entity.getProperty("comments");
       commentList.add(commentEntity);
```

Comment:
You might be able to optimize this code. Currently, it iterates through the result and stores it into an array (O(n)), then iterates over it again to return the result (another O(n)). You can instead combine the two steps into one single O(n) execution.